src/main/java/de/dhbw/karlsruhe/top/down/parsing/models/TopDownStep.java

```diff
@@ -0,0 +1,59 @@
+package de.dhbw.karlsruhe.top.down.parsing.models;
+
+import de.dhbw.karlsruhe.models.GrammarRule;
+
+public class TopDownStep {
+    private String readInput;
+    private TopDownState newState;
+    private String newStack;
+    private GrammarRule usedProduction;
```

Comment:
Die Variablennamen newState und newStack ergeben im Klassenkontext des TopDownStep keinen Sinn, da ein Step nur einen State und Stack hat. Dass der Step für den TopDownAcceptor immer ein neuer Step ist, sollte denke ich nicht Einfluss auf die TopDownStep Klasse haben.

Reply:
Ich habe mich da an die toString Methode gehalten. Dort wird es von mir auch als newState und newStack ausgegeben. Ich meine mich zu erinnern, dass wir das in der Vorlesung auch immer so genannt haben und ich die Namen deswegen wahrscheinlich so gewählt hab

Reply:
Wir können uns aber gerne nochmal über die Benennung von den Variablen unterhalten

Reply:
Wie unten schon erwähnt, ist die Funktionalität nicht gut in einen Methodennamen zu packen und wird deshalb in einem Kommentar erwähnt. Falls du einen andern Vorschlag hast, bin ich dafür natürlich offen

Reply:
Wir können gerne mal drüber reden. Ich sehe immer noch keinen Grund für die Benennung mit new. Wird zum Beispiel der ganze TopDownAcceptor ausgegeben, dann steht bei jedem Step newState. Wenn du hier oben die Variablennamen anpasst, dann natürlich auch in der toString() ;) Die wird ja automatisch anhand der Variablennamen generiert.